BigQuery/13_iv_antibiotics_pharmacy.sql

```diff
@@ -0,0 +1,86 @@
+
+select 
+subject_id,
+hadm_id,
+stay_id,
+DATETIME_TRUNC(starttime, HOUR) as chart_hour,
+intime,
+outtime,
+medication,
+-- todo: instead of taking the arbitrary minimum, take the values from the latest entertime for a given stay_id/chart_hour/medication combination 
+min(entertime) as entertime_min,
+min(starttime) as starttime_min,
+min(route) as route_min,
+max(stoptime) as stoptime_max,
+min(frequency) as frequency_min,
+min(disp_sched) as disp_sched_min,
+min(doses_per_24_hrs) as doses_per_24_hrs_min,
+min(duration) as duration_min,
+min(duration_interval) as duration_interval_min,
+count(1) as records
+from `physionet-data.mimic_hosp.pharmacy`
+JOIN `physionet-data.mimic_icu.icustays`
+USING (subject_id, hadm_id)
+where
+(
+    --medical abx names
+    lower(medication) like "%vancomycin%"
+    or lower(medication) like "%ceftriaxone%"
+    or lower(medication) like "%meropenem%"
+    or lower(medication) like "%ceftazidime%"
+    or lower(medication) like "%cefotaxime%"
+    or lower(medication) like "%cefepime%"
+    or lower(medication) like "%piperacillin%"
+    or lower(medication) like "%tazobactam%"
+    or lower(medication) like "%amipcillin%"
+    or lower(medication) like "%sulbactam%"
+    or lower(medication) like "%imipenem%"
+    or lower(medication) like "%cilastatin%"
+    or lower(medication) like "%levofloxacin%"
+    or lower(medication) like "%clindamycin%"
+    or lower(medication) like "%metronidazol%"
+    or lower(medication) like "%linezolid%"
+    or lower(medication) like "%ciprofloxacin%"
+    or lower(medication) like "%nafcillin%"
+    or lower(medication) like "%daptomycin%"
+    or lower(medication) like "%amikacin%"
+    or lower(medication) like "%tobramycin%"
+    or lower(medication) like "%gentamicin"
+    --brand names
+    or lower(medication) like "%vancocin%"
+    or lower(medication) like "%rocephin%"
+    or lower(medication) like "%merrem%"
+    or lower(medication) like "%fortaz%"
+    or lower(medication) like "%claforan%"
+    or lower(medication) like "%maxipime%"
+    or lower(medication) like "%zosyn%"
+    or lower(medication) like "%unasyn%"
+    or lower(medication) like "%primaxin%"
+    or lower(medication) like "%levaquin%"
+    or lower(medication) like "%cleocin%"
+    or lower(medication) like "%flagyl%"
+    or lower(medication) like "%zyvox%"
+    or lower(medication) like "%cetraxal%"
+    or lower(medication) like "%cipro%"
+    or lower(medication) like "%otiprio%"
+    or lower(medication) like "%ciloxan%"
+    or lower(medication) like "%cubicin%"
+    or lower(medication) like "%tobi podhaler%"
+    or lower(medication) like "%bethkis%"
+    or lower(medication) like "%tobrex%"
+    or lower(medication) like "%gentak%"
+)
+and route in (
+    'IV' --this captures by far the largest portion of abx orders, followed with a few from IVIC and IVT. The others are here for posterity
+    ,'IVCI'
+    ,'IVIC'
+    ,'IVS'
+    ,'IVT')
```

Comment:
Are there false positives for matching on `IV%`?

Reply:
Also noticed that this is not using the lowercase like other spots. Not an issue, I think, but then I'm not sure about the source of these data, either. Enumerating them is definitely safer than matching on `IV%`. May not need to be anything done here except lowercasing for consistency, but that's a nit.

Reply:
Yes, there are false positives for matching on IV. The reason why I didn't want to use lowercase is because we're doing exact matching on the available fields in the DB, but I do acknowledge we lose consistency since the rest of the queries are using "like" filters on lower(field)

Reply:
Oy vey - of course there are things that start with "IV" that have nothing to do with IVs! Being specific on ETL queries is probably going to be the trend.

I hate to harp on consistency, but it might make sense for us to even just transform to lowercase for everything so that it doesn't have to pepper everything we do. What do you think?